src/source/Signaling/Signaling.c

```diff
@@ -434,12 +461,14 @@ STATUS refreshIceConfigurationCallback(UINT32 timerId, UINT64 scheduledTime, UIN
 
     DLOGD("Refreshing the ICE Server Configuration");
 
-    // Check if we are in a connected state and if not bail
+    // Check if we are in a connected state or ready state and if not bail.
+    // The ICE state will be called in any other states
     CHK_STATUS(getStateMachineCurrentState(pSignalingClient->pStateMachine, &pStateMachineState));
-    CHK(pStateMachineState->state == SIGNALING_STATE_CONNECTED, retStatus);
+    CHK(pStateMachineState->state == SIGNALING_STATE_CONNECTED ||
+        pStateMachineState->state == SIGNALING_STATE_READY, retStatus);
 
     // Force terminate the connection for now as LWS doesn't allow parallel processing
-    terminateConnectionWithStatus(pSignalingClient, SERVICE_CALL_RESULT_SIGNALING_RECONNECT_ICE);
+    ATOMIC_STORE(&pSignalingClient->result, (SIZE_T) SERVICE_CALL_RESULT_SIGNALING_RECONNECT_ICE);
```

Comment:
It looks like there are lots of signals passed around using the pSignalingClient ->result variable. Any chance it can get overridden? If so are we ok with that?

Reply:
The state machine that's checking these have a serialized access